src/Csv.php

```diff
@@ -1310,4 +1311,48 @@ protected function _guess_delimiter($search_depth, $preferred, $enclosure, &$dat
         ksort($filtered);
         $this->delimiter = reset($filtered);
     }
+
+    /**
+     * getCollection
+     * Returns a Illuminate/Collection object
+     * This may prove to be helpful to people who want to
+     * create macros, and or use map functions
+     *
+     * @access public
+     * @link   https://laravel.com/docs/5.6/collections
+     *
+     * @throws ErrorException - If the Illuminate\Support\Collection class is not found
+     * @throws MixedException - If the new Collection class throws an exception
+     * @throws ErrorException - If the collection object is null after initing the collection class
+     *
+     * @return Illuminate\Support\Collection
+     */
+    public function getCollection() {
+        //start a null collection object - we will check this later
+        $collection = null;
+
+        //does the Illuminate\Support\Collection class exists?
+        //this uses the autoloader to try to determine
+        //@see http://php.net/manual/en/function.class-exists.php
+        if (class_exists('Illuminate\Support\Collection', true) == false) {
+            throw new \ErrorException('It would appear you have not installed the illuminate/support package!');
+        }
+
+        //now try creating the collection
+        try {
+            $collection = new Collection($this->data);
+        } catch (\Exception $e) {
+            //if an exception was raised rethrow it
+            throw $e;
```

Comment:
1. Rethrowing seems unnecessary to me, as I don't think it adds any value.
2. From the Collection class source code and my few experiments I couldn't find any situation in which it would throw the exception. Do you recall when it is necessary?

Reply:
I'm always writing code that traps errors (even if they don't exists). I'll remove this shortly.

Reply:
@williamknauss A very good practice indeed. I should have used more words to explain myself: Exception handling is a valuable thing to do. Catching an exception and re-throwing it under all conditions without any value-add was what caught my eye.

Reply:
@Fonata  - Totally understood.